gslab_scons/tests/test_size_warning.py

```diff
@@ -0,0 +1,312 @@
+             'It took 2.44 seconds to enumerate untracked files. \'status -uno\'\n',
+             'may speed it up, but you have to be careful not to forget to add\n',
+             'new files yourself (see \'git help status\').\n',
+             'no changes added to commit (use "git add" and/or "git commit -a")\n']
+        
+        # ii) No files are ignored
+        none_ignored = \
+            ['On branch issue59-size_warnings\n',
+             'Your branch is up-to-date with \'origin/issue59-size_warnings\'.\n',
+             'nothing to commit, working tree clean\n']
+
+        if ignored == 'none':
+            message = ''.join(none_ignored)
+        else:
+            message = ''.join(standard)
+
+        command = args[0]
+        if 'shell' in kwargs.keys():
+            if kwargs['shell'] and (command == 'git status --ignored'):
+                return message
+    
+        return None
+
+    return effect
+
+
+def walk_ignored_side_effect(*args, **kwargs):
+    '''Mock os.walk() for testing list_ignored_files()'''
+    path = args[0]
+    path = os.path.relpath(path)
+
+    if path not in struct.keys():
+        raise StopIteration
+
+    if path == 'release':
+        subdirs = ['subdir']
+    else:
+        subdirs = []
+
+    yield (path, subdirs, struct[path])
+ 
+
+def isdir_ignored_side_effect(*args, **kwargs):
+    path = args[0]
+    if path == '':
+        return False
+        
+    isdir = (os.path.relpath(path) in struct.keys())
+    return isdir
+
+
+def isfile_ignored_side_effect(*args, **kwargs):
+    path = args[0]
+    if path == '':
+        return False
+
+    file_list = []
+
+    for k in struct.keys():
+        file_list += [os.path.join(k, f) for f in struct[k]]
+
+    isfile = (os.path.relpath(path) in map(os.path.relpath, file_list))
+    return isfile
+
+
+#== Side effects for testing create_size_dictionary() ===
+# These functions mock a directory containing files of various sizes
+# and a system that recognises no directory other than that one.
+def isdir_dict_side_effect(*args, **kwargs):
+    '''
+    Mock os.path.isdir() so that it only recognises ./test_files/
+    as an existing directory.
+    '''
+    path = args[0]
+    if not isinstance(path, str):
+        raise TypeError('coercing to Unicode: need string or buffer, '
+                        '%s found' % type(path))
+    return path in ['test_files', 'release']
+
+
+def walk_dict_side_effect(*args, **kwargs):
+    '''Mock os.walk() for a mock directory called ./test_files/.'''
+    path = args[0]
+
+    # Mock two directories at the root, one of which has a subdirectory
+    roots       = ['test_files',      'test_files/size_test',   'release']
+    directories = [['size_test'],     [],                       []]
+    files       = [['root_file.txt'], ['test.txt', 'test.pdf'], ['output.txt']]
+
+    for i in range(len(roots)):
+        if re.search('^%s' % path, roots[i]):
+            yield (roots[i], directories[i], files[i])
```

Comment:
Is this so `path = test_files` also returns the contents of `test_files/size_test`? A comment might be good.